src/ChunkSplitters.jl

```diff
@@ -74,51 +74,73 @@ is_chunkable(::Tuple) = true
 # Current chunks split types
 const split_types = (:batch, :scatter)
 
+# User defined constraint
+abstract type Constraint end
+struct FixedCount <: Constraint end
+struct FixedSize <: Constraint end
+
 # Structure that carries the chunks data
-struct Chunk{T}
+struct Chunk{T,C<:Constraint}
     itr::T
     n::Int
+    size::Int
     split::Symbol
 end
 is_chunkable(::Chunk) = true
 
 # Constructor for the chunks
-function chunks(data; n::Int, split::Symbol=:batch)
-    n >= 1 || throw(ArgumentError("n must be >= 1"))
+function chunks(data; n::Integer=0, size::Integer=0, split::Symbol=:batch)
+    n != 0 || size != 0 || missing_input_err()
+    n != 0 && size != 0 && mutually_exclusive_err()
+    if n != 0
+        C = FixedCount
+        n >= 1 || throw(ArgumentError("n must be >= 1"))
+    else
+        C = FixedSize
+        size >= 1 || throw(ArgumentError("size must be >= 1"))
+        size <= length(data) || throw(ArgumentError("size must be <= length(data)"))
+    end
     is_chunkable(data) || not_chunkable_err(data)
     (split in split_types) || split_err()
-    Chunk{typeof(data)}(data, min(length(data), n), split)
+    Chunk{typeof(data),C}(data, min(length(data), n), size, split)
```

Comment:
Do we need these tests here? Aren't the input errors inside `getchunk` not redundant?

Reply:
At least maybe we can split the tests into a function and call it in both places, to not repeat the code.